jib-cli/src/main/java/com/google/cloud/tools/jib/cli/cli2/Containerizers.java

```diff
@@ -0,0 +1,147 @@
+import com.google.cloud.tools.jib.global.JibSystemProperties;
+import com.google.cloud.tools.jib.plugins.common.logging.ConsoleLogger;
+import com.google.cloud.tools.jib.plugins.common.logging.ProgressDisplayGenerator;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+/** Helper class for creating Containerizers from JibCli specifications. */
+public class Containerizers {
+
+  /**
+   * Create a Containerizer from a jibcli command line specification.
+   *
+   * @param buildOptions JibCli options
+   * @return a populated Containerizer
+   * @throws InvalidImageReferenceException if the image reference could not be parsed
+   */
+  public static Containerizer from(JibCli buildOptions) throws InvalidImageReferenceException {
+    ConsoleLogger logger =
+        CliLogger.newLogger(buildOptions.getVerbosity(), buildOptions.getConsoleOutput());
+
+    Containerizer containerizer = create(buildOptions, logger);
+
+    applyLogger(containerizer, logger);
+    applyConfiguration(containerizer, buildOptions);
+
+    return containerizer;
+  }
+
+  static Containerizer create(JibCli buildOptions, ConsoleLogger logger)
+      throws InvalidImageReferenceException {
+    String imageSpec = buildOptions.getTargetImage();
+    if (imageSpec.startsWith(DOCKER_DAEMON_IMAGE_PREFIX)) {
+      // TODO: allow setting docker env and docker executable (along with path/env)
+      return Containerizer.to(
+          DockerDaemonImage.named(imageSpec.replaceFirst(DOCKER_DAEMON_IMAGE_PREFIX, "")));
+    }
+    if (imageSpec.startsWith(TAR_IMAGE_PREFIX)) {
+      return Containerizer.to(
+          TarImage.at(Paths.get(imageSpec.replaceFirst(TAR_IMAGE_PREFIX, "")))
+              .named(buildOptions.getName()));
+    }
+    ImageReference imageReference =
+        ImageReference.parse(imageSpec.replaceFirst(REGISTRY_IMAGE_PREFIX, ""));
+    RegistryImage registryImage = RegistryImage.named(imageReference);
+    applyCredentialConfig(
+        registryImage,
+        CredentialRetrieverFactory.forImage(
+            imageReference, logEvent -> logger.log(logEvent.getLevel(), logEvent.getMessage())),
+        buildOptions);
+    return Containerizer.to(registryImage);
+  }
+
+  static void applyCredentialConfig(
+      RegistryImage registryImage, CredentialRetrieverFactory factory, JibCli buildOptions) {
+    if (buildOptions.getUsernamePassword().isPresent()) {
+      registryImage.addCredentialRetriever(buildOptions::getUsernamePassword);
+    } else if (buildOptions.getToUsernamePassword().isPresent()) {
+      registryImage.addCredentialRetriever(buildOptions::getToUsernamePassword);
+    }
+    for (String credentialHelper : buildOptions.getCredentialHelpers()) {
+      Path path = Paths.get(credentialHelper);
+      if (Files.exists(path)) {
+        registryImage.addCredentialRetriever(factory.dockerCredentialHelper(path));
+      } else {
```

Comment:
I think this needs a little more work for Windows (as in `DefaultCredentialRetrievers.asList()`). Add a TODO?

Reply:
oh didn't know about this, basically just pulled this out of brian's code. I can just use that instead.

Reply:
Yeah, Brian must have copied the old `DefaultCredentialRetrievers` code, and then we fixed some Windows issues not long ago.